otherlibs/stdune/ansi_color.ml

```diff
@@ -239,15 +415,21 @@ let parse_line str styles =
             else
               String.sub str ~pos:seq_start ~len:(seq_end - seq_start)
               |> String.split ~on:';'
+              |> List.map ~f:Style.of_ansi_string
```

Comment:
Could you move the conversion into the fold and remove this extra pass on the list?